<?xml version="1.0" encoding="UTF-8"?>
<launch>
      <arg name="plot" default="false"/>
      <arg name="enable_obstacle" default="true"/>
      <arg name="debug" default="false"/>

      <arg name="experiment" default="false"/>
      <arg name="experiment_id" default="manipulation_target_reaching"/>

      <!-- Open Rviz -->
      <node name="rviz" pkg="rviz" type="rviz" args="-d $(find husky_panda_manipulation)/config/config.rviz" output="screen"/>

      <!--    Plot cost and other info-->
      <group if="$(arg plot)">
            <arg name="plot_config" default="$(find husky_panda_manipulation)/config/plotjuggler.xml"/>
            <node name="plotjuggler" pkg="plotjuggler" type="plotjuggler" args="--layout $(arg plot_config)" output="screen"/>
      </group>

      <!--    Interactive marker to command the end effector-->
      <node name="interactive_reference" pkg="mppi_panda" type="interactive_marker.py" output="screen">
            <param name="marker_server_name" value="reference_marker_server"/>
            <param name="subscribe_initial_pose" value="true"/>
            <param name="initial_pose_topic" value="/end_effector"/>
            <param name="frame_id" value="world"/>
            <param name="target_pose_topic" value="/end_effector_pose_desired"/>
      </node>

      <!--    Interactive marker for the end effector obstacle -->
      <group if="$(arg enable_obstacle)">
            <node name="interactive_obstacle" pkg="mppi_panda" type="interactive_marker.py" output="screen">
                  <param name="marker_server_name" value="obstacle_marker_server"/>
                  <param name="subscribe_initial_pose" value="false"/>
                  <param name="frame_id" value="world"/>
                  <param name="target_pose_topic" value="/obstacle"/>
            </node>
      </group>

      <group if="$(arg experiment)">
            <node pkg="husky_panda_ros" type="record_data.py" name="record_data">
                  <param name="experiment_id" value="$(arg experiment_id)"/>
            </node>
            <node pkg="husky_panda_manipulation" type="target_generator.py" name="target_generator" output="screen">
                  <param name="experiment_id" value="panda_mobile_kin_multi_target"/>
            </node>
      </group>
</launch>